libs/ShaderProgram/ShaderProgram.h

```diff
@@ -0,0 +1,48 @@
+#ifndef SHADER_PROGRAM_H
+#define SHADER_PROGRAM_H
+
+#include <GL/glew.h>
+
+#define GLM_ENABLE_EXPERIMENTAL
+#include <glm/glm.hpp>
+#include <glm/gtx/transform.hpp>
+#include <glm/gtc/type_ptr.hpp>
+#include <glm/gtx/quaternion.hpp>
+#include <glm/gtc/quaternion.hpp>
+
+#include <unordered_map>
+
+using namespace glm;
+
+
+class ShaderProgram
+{
+
+public:
+	ShaderProgram();
+	void LoadShaders(const char* vertex_filenname, const char* fragment_filename);
+	~ShaderProgram();
+	void start();
+	void stop();
+
+	void addUniform(const char* name);
+
+	void setUniformi(const char* name, int value);
+	void setUniformf(const char* name, float value);
+	void setUniform(const char* name, vec3 vector);
```

Comment:
Pass vector by reference?

Reply:
Fixed. Also, removed `using namespace` and added `glm::`